### PR TITLE
Durable Storage ADR

### PR DESCRIPTION
**Pull request by lalexgap**
This adds an ADR specifying the details of why we selected `BuntDB` to implement our persistent store.

### CODE REVIEW

**Review by geoknee — APPROVED** [review]
Thanks @lalexgap!

The overall diff includes many source code files, too. So let's make sure to pinch those out before we merge this.

**Comment by lalexgap** [comment]
> Thanks @lalexgap!
> 
> The overall diff includes many source code files, too. So let's make sure to pinch those out before we merge this.

Ah good catch, I forgot to update this PR after some rebases upstream.

**Comment by geoknee** [comment]
Could we hold of merging for a little while, I just want to coordinate with some other ADR PRs we have open.

File: .adr/0011-persistent-storage.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# 0011 Implementing a Persistent Store
```
Comment (geoknee) [inline .adr/0011-persistent-storage.md:1]:
```suggestion
# 0011 Implementing a Durable Store
```

**Pull request merged**